### DISCUSSION TITLE
CORS

### DISCUSSION THREAD

**Discussion by juanbautista0** [clue/framework-x#99] [Q&A] [answered]
in the constructor of the _**React\Http\Message\Response**_ object can CORS be enabled?

**Comment by clue** [comment, marked as answer]
Excellent question! CORS headers are supported just like any other response headers.

```php
$app->get('/user', function () {
    return new React\Http\Message\Response(
        200,
        [
            'Content-Type' => 'text/plain; charset=utf-8',
            'Access-Control-Allow-Origin' => '*'
        ],
        "Hello wörld"
    );
});
```

See https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS and https://framework-x.org/docs/api/response/#headers.

Depending on your use case, you may add these headers for each route or use a middleware implementation that can be reused for a number of routes.
